### DISCUSSION TITLE
Can you customize the memory size of jdtls?

### DISCUSSION THREAD

**Discussion by xuliefeng** [williamboman/nvim-lsp-installer#506] [Q&A] [answered]
hi，Can you customize the memory size of jdtls? I want to increase the memory value

**Comment by williamboman** [comment]
Hello! At the moment, in order to do this you'll need to override the `on_new_config` settings and provide your own `cmd`. Maybe something like this would work:

```lua
lsp_installer.on_server_ready(function (server)
    local opts = {}
 
    if server.name == "jdtls" then
        local on_new_config = server:get_default_options().on_new_config
        opts.on_new_config = function (config, workspace_dir)
            on_new_config(config, workspace_dir)
            table.insert(config.cmd, "-Xms1g")
            table.insert(config.cmd, "-Xmx2g")
        end
    end   
 
    server:setup(opts)
end)
```

**Reply by xuliefeng** [reply]
OK  Thanks

**Comment by xuliefeng** [comment]
not working but value is insert to cmd,  Maybe it's a parameter order issue

**Comment by williamboman** [comment, marked as answer]
Hello again! This should now be easier to accomplish by customizing the `vmargs` configuration, more information in this README: https://github.com/williamboman/nvim-lsp-installer/blob/main/lua/nvim-lsp-installer/servers/jdtls/README.md#customizing-jvm-arguments
